bot/cogs/moderation/silence.py

```diff
@@ -76,6 +47,21 @@ async def _get_instance_vars(self) -> None:
         self._mod_log_channel = self.bot.get_channel(Channels.mod_log)
         self._get_instance_vars_event.set()
 
+    async def schedule_unsilence(self, channel: SilencedChannel) -> None:
+        """Schedule expiration for silenced channels."""
+        await self.bot.wait_until_guild_available()
+        log.debug("Scheduling unsilencer")
+        self.schedule_task(channel.id, channel)
+
+    async def _scheduled_task(self, channel: SilencedChannel) -> None:
+        """Calls `self.unsilence` on expired silenced channel to unsilence it."""
+        await time.wait_until(channel.stop)
```

Comment:
`time.wait_until` requires datetime which shouldn't be required here, it would be easier to just pass the number of minutes into `asyncio.sleep` (it's more convenient to convert minutes into seconds here rather than outside).